File: Source/Parser/IArguable.cs
using System.Collections.Generic;

namespace Pyratron.Frameworks.Commands.Parser
{
    /// <summary>
    /// Represents an object that has arguments/parameters
    /// </summary>
    internal interface IArguable
    {
        /// <summary>
        /// The input (Including alias and help) that are passed with the command or argument.
        /// </summary>
        List<Argument> Arguments { get; set; }
    }
}

File: Java/Parser/src/com/pyratron/frameworks/commands/parser/IArguable.java
package com.pyratron.frameworks.commands.parser;

import java.util.ArrayList;

/**
 * Represents an object that has multiple arguments/parameters.
 */
public interface IArguable // Calm down! Why are we arguing?!
{
    /**
     * The arguments the object contains. Arguments may be nested inside others to create links of arguments.
     */
    ArrayList<Argument> getArguments();
}
